plugin/src/test/java/org/opensearch/ml/rest/RestMLExecuteActionTests.java

```diff
@@ -281,4 +282,59 @@ public void testPrepareRequestSystemException() throws Exception {
+                );
+            return null;
+        }).when(client).execute(eq(MLExecuteTaskAction.INSTANCE), any(), any());
+        doNothing().when(channel).sendResponse(any());
+        when(channel.newBuilder()).thenReturn(XContentFactory.jsonBuilder());
+        restMLExecuteAction.handleRequest(request, channel, client);
+
+        ArgumentCaptor<MLExecuteTaskRequest> argumentCaptor = ArgumentCaptor.forClass(MLExecuteTaskRequest.class);
+        verify(client, times(1)).execute(eq(MLExecuteTaskAction.INSTANCE), argumentCaptor.capture(), any());
+        Input input = argumentCaptor.getValue().getInput();
+        assertEquals(FunctionName.AGENT, input.getFunctionName());
+        ArgumentCaptor<RestResponse> restResponseArgumentCaptor = ArgumentCaptor.forClass(RestResponse.class);
+        verify(channel, times(1)).sendResponse(restResponseArgumentCaptor.capture());
+        BytesRestResponse response = (BytesRestResponse) restResponseArgumentCaptor.getValue();
+        assertEquals(RestStatus.BAD_REQUEST, response.status());
+        assertEquals("application/json; charset=UTF-8", response.contentType());
+        String expectedError =
+            "{\"status\":400,\"error\":{\"type\":\"IllegalArgumentException\",\"reason\":\"Invalid Request\",\"details\":\"Illegal Argument Exception\"}}";
+        assertEquals(expectedError, response.content().utf8ToString());
+    }
+
+    public void testAgentExecutionResponsePlainText() throws Exception {
+        RestRequest request = getExecuteAgentRestRequest();
+        doAnswer(invocation -> {
+            ActionListener<MLExecuteTaskResponse> actionListener = invocation.getArgument(2);
+            actionListener
+                .onFailure(
+                    new RemoteTransportException("Remote Transport Exception", new IllegalArgumentException("Illegal Argument Exception"))
+                );
+            return null;
+        }).when(client).execute(eq(MLExecuteTaskAction.INSTANCE), any(), any());
+        doNothing().when(channel).sendResponse(any());
+        restMLExecuteAction.handleRequest(request, channel, client);
+
+        ArgumentCaptor<MLExecuteTaskRequest> argumentCaptor = ArgumentCaptor.forClass(MLExecuteTaskRequest.class);
+        verify(client, times(1)).execute(eq(MLExecuteTaskAction.INSTANCE), argumentCaptor.capture(), any());
+        Input input = argumentCaptor.getValue().getInput();
+        assertEquals(FunctionName.AGENT, input.getFunctionName());
+        ArgumentCaptor<RestResponse> restResponseArgumentCaptor = ArgumentCaptor.forClass(RestResponse.class);
+        verify(channel, times(1)).sendResponse(restResponseArgumentCaptor.capture());
+        BytesRestResponse response = (BytesRestResponse) restResponseArgumentCaptor.getValue();
+        assertEquals(RestStatus.BAD_REQUEST, response.status());
+        assertEquals("text/plain; charset=UTF-8", response.contentType());
```

Comment:
Difference between these two tests are only `assertEquals("application/json; charset=UTF-8", response.contentType());` vs `assertEquals("text/plain; charset=UTF-8", response.contentType());`. Is that expected?

Reply:
Yes, it is expected. Actually there is another difference `when(channel.newBuilder()).thenReturn(XContentFactory.jsonBuilder());`
Without this jsonBuilder, the xcontent building will fail, then downgrade to catch branch to send plain text response.

Reply:
[nit] I see. In that case, what do you think about something like following:

```
public void testAgentExecutionResponse() throws Exception {
    // Test JSON response
    runAgentExecutionResponseTest(
        "application/json; charset=UTF-8",
        "{\"status\":400,\"error\":{\"type\":\"IllegalArgumentException\",\"reason\":\"Invalid Request\",\"details\":\"Illegal Argument Exception\"}}",
        true
    );

    // Test plain text response
    runAgentExecutionResponseTest(
        "text/plain; charset=UTF-8",
        "{\"error\":{\"reason\":\"Invalid Request\",\"details\":\"Illegal Argument Exception\",\"type\":\"IllegalArgumentException\"},\"status\":400}",
        false
    );
}

private void runAgentExecutionResponseTest(String expectedContentType, String expectedError, boolean isJsonResponse) throws Exception {
    RestRequest request = getExecuteAgentRestRequest();
    
    // Mock client behavior
    doAnswer(invocation -> {
        ActionListener<MLExecuteTaskResponse> actionListener = invocation.getArgument(2);
        actionListener.onFailure(new RemoteTransportException("Remote Transport Exception", new IllegalArgumentException("Illegal Argument Exception")));
        return null;
    }).when(client).execute(eq(MLExecuteTaskAction.INSTANCE), any(), any());
    
    doNothing().when(channel).sendResponse(any());

    // Conditionally mock JSON response building
    if (isJsonResponse) {
        when(channel.newBuilder()).thenReturn(XContentFactory.jsonBuilder());
    }

    restMLExecuteAction.handleRequest(request, channel, client);

    // Verify request processing
    ArgumentCaptor<MLExecuteTaskRequest> argumentCaptor = ArgumentCaptor.forClass(MLExecuteTaskRequest.class);
    verify(client, times(1)).execute(eq(MLExecuteTaskAction.INSTANCE), argumentCaptor.capture(), any());
    Input input = argumentCaptor.getValue().getInput();
    assertEquals(FunctionName.AGENT, input.getFunctionName());

    // Verify response
    ArgumentCaptor<RestResponse> restResponseArgumentCaptor = ArgumentCaptor.forClass(RestResponse.class);
    verify(channel, times(1)).sendResponse(restResponseArgumentCaptor.capture());
    BytesRestResponse response = (BytesRestResponse) restResponseArgumentCaptor.getValue();
    assertEquals(RestStatus.BAD_REQUEST, response.status());
    assertEquals(expectedContentType, response.contentType());
    assertEquals(expectedError, response.content().utf8ToString());
}
```

Reply:
Yeah, this looks good. Actually I would prefer to refactor all tests in `RestMLExecuteActionTests` in next pr, as some existing test cases have the similar improvement to do.

Reply:
Sure, approving the PR in expectation of a follow up PR to refactor the tests in `RestMLExecuteActionTests`